app/helpers/application_helper.rb

```diff
@@ -170,6 +170,14 @@ def copyright_notice
     "#{start_year} - #{current_year}"
   end
 
+  def mail_link(type = :default, text: nil, additional_info: nil)
+    email = SiteConfig.email_addresses[type] || SiteConfig.email_addresses[:default]
+    href = "mailto:#{email}#{"?#{additional_info}" if additional_info}"
```

Comment:
what if additional info is a key value param you parse? this way it can be used as:

```ruby
(e)mail_link(:default, text: "abc", additional_info: {subject: "Title"})
```

and inside you do:

```ruby
[5] pry(main)> URI.encode_www_form(subject: "This is a long subject", body: "This is a longer body with a question mark ?")
=> "subject=This+is+a+long+subject&body=This+is+a+longer+body+with+a+question+mark+%3F"
``` 

currently it's not going to escape the chars like space and question mark, which would, for the latter case, result in a double question mark:

```ruby
[8] pry(main)> additional_info = URI.encode_www_form(subject: "This is a long subject", body: "This is a longer body with a question mark ?")
=> "subject=This+is+a+long+subject&body=This+is+a+longer+body+with+a+question+mark+%3F"
[9] pry(main)> content_tag(:a, "text", {href: "mailto:a@bc.com?#{additional_info}"})
=> "<a href=\"mailto:a@bc.com?subject=This+is+a+long+subject&amp;body=This+is+a+longer+body+with+a+question+mark+%3F\">text</a>"
[11] pry(main)> content_tag(:a, "text", {href: "mailto:a@bc.com?subject=Is this it?"})
=> "<a href=\"mailto:a@bc.com?subject=Is this it?\">text</a>"
```

Reply:
@rhymes this is good, and I like the premise of this. I did find a caveat though - URI.encode_www_form will break on spaces because the `+` is used to encode spaces in URLs, not in HTML and not in SMTP.  So for example the apple mail app would render the subject spaces as `This+is+a+long+subject`.  

However I did take inspiration from [Rail's mail_to](https://apidock.com/rails/ActionView/Helpers/UrlHelper/mail_to) and used ERB::Util.url_encode along with some other nice details for code safety that it has  (like deleting sanitizing the additional parameters.

Reply:
Nice work! BTW, why can't we simply use mail_to internally instead of reimplementing it?

Reply:
Something like:

```ruby
def email_link(type = :default, text: nil, additional_info: nil)
  # extract email address(es)
  # call mail_to
end
```

Reply:
@rhymes I was literally just implementing that now, when I typed out that comment I thought why in the world am I reimplementing mail_to when we can use it 🙈

Reply:
Lol what a round about way to get here but I think we may have finally got it! 🎉